File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty())
            return "";

        string ans = "";
        int minLength;

        // Find the minimum length of all strings in the vector
        for (string s : strs) {
            minLength = min(minLength, static_cast<int>(s.size()));
        }

        // Compare characters at the same position in all strings
        for (int i = 0; i < minLength; i++) {
            char currentChar = strs[0][i];
            for (int j = 1; j < strs.size(); j++) {
                if (strs[j][i] != currentChar) {
                    // Found a mismatch, return the current common prefix
                    return ans;
                }
            }
            ans += currentChar;
        }

        return ans;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = ""

        min_word = min(strs)
        max_word = max(strs)

        length = min(len(min_word), len(max_word))
        
        i = 0
        while (i < length):
            if (min_word[i] == max_word[i]):
                ans += min_word[i]
            else:
                return ans
            i += 1

        return ans